### DISCUSSION TITLE
How can I enable pointer events in custom controls?

### DISCUSSION THREAD

**Discussion by KevinMorales0** [microsoft/microsoft-ui-xaml#8138] [Q&A] [answered]
I am tried to create a new control. But the attached pointer events are not triggered. 

Pseudo example: 

Class CustomButton: Control
{
    public CustomButton()
    {
       Height = 200
       Width = 200
       Background = SolidColorBrush(colors.Red)
       PointerEntered += MagicAction;
    }
    public void MagicAction(object sender, PointerEventHandlerArgs e)
    {
        ...
    }
}

**Comment by castorix** [comment]
It works for me with 

        private void MagicAction(object sender, PointerRoutedEventArgs e)
        {
            // Code...
        }

**Reply by KevinMorales0** [reply]
hi @castorix,
What process do you use to create the control object?

**Comment by KevinMorales0** [comment]
Update:
When I use the winui Template, the new control works. But When I create a new class that inherit the class control, It doesn't Works.

Now, Why this happen?

**Comment by KevinMorales0** [comment, marked as answer]
Okay, I have a solution.

Use a Border as a wrapper and pass the events to it.
